reactive/etcd.py

```diff
@@ -97,13 +105,53 @@ def update_cluster_string():
 
 @when_not('etcd.installed')
 def install_etcd():
-    source = config('source-url')
-    sha = config('source-sum')
-
     status_set('maintenance', 'Installing etcd.')
-    etcd_helper = EtcdHelper()
-    etcd_helper.fetch_and_install(source, sha)
-    set_state('etcd.installed')
+
+    codename = host.lsb_release()['DISTRIB_CODENAME']
+
+    etcd_path = resource_get('etcd')
+    etcdctl_path = resource_get('etcdctl')
+
+    if not etcd_path or not etcdctl_path:
+        if codename == 'xenial':
+            # edge case where archive allows us a nice fallback on xenial
+            status_set('maintenance', 'Attempting install of etcd from apt')
+            pkg_list = ['etcd']
+            apt_update()
+            apt_install(pkg_list, fatal=True)
+            set_state('etcd.installed')
+            return
+        else:
+            # edge case
+            status_set('blocked', 'Missing Resource: see README')
+    else:
+        install(etcd_path, '/usr/bin/etcd')
+        install(etcdctl_path, '/usr/bin/etcdctl')
+
+        host.add_group('etcd')
+
+        if not host.user_exists('etcd'):
+            host.adduser('etcd')
+            host.add_user_to_group('etcd', 'etcd')
+
+        os.makedirs('/var/lib/etcd/', exist_ok=True)
```

Comment:
Thats as fair point, but I'd rather let python 2.7 silently die off. Charms  have migrated to python3 since we started the layer approach. I'd like to keep pushing the envelope of making the transition.